Covinator/Covinator/Pages/PfizerVaccination.cshtml.cs

```diff
@@ -21,92 +19,112 @@ public PfizerVaccination(ILogger<PfizerVaccination> logger)
             _logger = logger;
         }
 
+        public IList<string> JurisdictionsList = new List<string>();
+
         public void OnGet()
         {
-            using (var webClient = new WebClient())
+            try
             {
-                string pfizerData = webClient.DownloadString("https://data.cdc.gov/resource/saz5-9hgg.json");
-                JSchema schema = JSchema.Parse(System.IO.File.ReadAllText("pfizerSchema.json"));
-                JArray jsonArray = JArray.Parse(pfizerData);
-                IList<string> validationEvents = new List<string>();
-                var currentWeek1 = jsonArray[0].ToArray();
-                var currentWeek = currentWeek1[1].ToString();
-                if (jsonArray.IsValid(schema, out validationEvents))
+                using (var webClient = new WebClient())
                 {
+                    string pfizerData = webClient.DownloadString("https://data.cdc.gov/resource/saz5-9hgg.json");
+                    JSchema schema = JSchema.Parse(System.IO.File.ReadAllText("pfizerSchema.json"));
+                    JArray jsonArray = JArray.Parse(pfizerData);
+                    IList<string> validationEvents = new List<string>();
 
-                    JArray result_array = new JArray();
-                    JArray jarray = new JArray();
-                    var length_arr = jsonArray.Count();
-                    for (int i=0; i<63; i++)
-                    {
-                        result_array.Add(jsonArray[i]);
-
-                    }
+                    var currentWeek = jsonArray.FirstOrDefault()?["week_of_allocations"]?.Value<DateTime>().ToShortDateString();
 
-                    string result_string = result_array.ToString();
+                    if (jsonArray.IsValid(schema, out validationEvents))
+                    {
+                        JArray result_array = new JArray();
 
-                    var pfizerVaccineDistributionAllocations = PfizerVaccineDistributionAllocations.FromJson(result_string);
+                        for (int i = 0; i < 63; i++)
+                        {
+                            result_array.Add(jsonArray[i]);
+                        }
 
-                    /*ViewData["PfizerVaccineDistributionAllocations"] = pfizerVaccineDistributionAllocations;*/
-                    ViewData["PfizerVaccineDistributionAllocations"] = pfizerVaccineDistributionAllocations;
-                    ViewData["CurrentWeek"] = currentWeek;
+                        var pfizerVaccineDistributionAllocations = PfizerVaccineDistributionAllocations.FromJson(result_array.ToString());
 
-                }
-                else
-                {
+                        PopulateJurisdictions(jsonArray);
```

Comment:
This calls a private method which populates a list containing all jurisdictions- this dynamically populates the drop down